pwndbg/typeinfo.py

```diff
@@ -119,18 +120,36 @@ def load(name):
 {name} foo;
 '''.format(**locals())
 
-    filename = '%s/%s_%s' % (tempdir, arch, '-'.join(name.split()))
+    filename = '%s/%s_%s.cc' % (tempdir, arch, '-'.join(name.split()))
 
-    if not os.path.exists(filename + '.o'):
-        with open(filename + '.cc', 'w+') as f:
-            f.write(source)
-            f.flush()
+    with open(filename, 'w+') as f:
+        f.write(source)
+        f.flush()
```

Comment:
Looks like for 2.7, `file.flush` invokes the native routine `file_flush` which invokes `fflush`.

`file.close` invokes the native routine `file_close` --> `close_the_file` --> `PyFileObject->f_close` --> `close(2)`.  There is no call to any flush routine.

`file.__exit__` just invokes the `close` method on the `file` object.

According to StackOverflow, `file.close()` implies `file.flush()`, but I don't see any code for it in the `cpython` source.

Separately, `flush()` does not guarantee that the data is actually on disk.  That requires calling `fsync()` or using `O_SYNC`.

So it appears that the call to `f.flush()` is both unnecessary *and* incorrect.